### ISSUE TITLE
Question about pre-processor

### ISSUE THREAD

**Issue by mattmazzola** [neiltownsley/react-typescript-cypress#2]
How is cypress aware of this preprocessor?  Does using .ts files inside the fixtures, plugins, and support folders work?  Does it work by filename convention? (`cy-ts-preprocessor` is special?)

This repo looks like it would have the same issue i'm facing which is that you can get the Cypress types by simply be adding the `tsconfig.json` but once you actually use typescript in the tests it will fail as a syntax error.  For example add something like:
```
const myVar: string = "abc"
```
The `:` is not valid syntax in js.  Without type declarations Cypress can use the .ts files as-is and it happens to work, but isn't as helpful as true TypeScript compilation.

In order to get the preprocessor I thought you would need to update the `plugins/index.ts` file to include: `on('file:preprocessor', wp(options))`

Info on the preprocessor event: https://github.com/cypress-io/cypress-example-recipes/blob/master/examples/preprocessors__typescript-webpack/cypress/plugins/index.js#L7

**Comment by rkrisztian** [comment]
I experimented with this and found the following.

By default, only JavaScript is a valid syntax, regardless of the file extension (`.js` or `.ts`).

The file `cypress/plugins/index.ts` seems to be the entry point of plugins loaded by Cypress. Thus, I put the following lines into that file:

```
const wp = require('@cypress/webpack-preprocessor');

module.exports = (on, config) => {
	on('file:preprocessor', wp({
		webpackOptions: {
			resolve: {
				extensions: ['.ts', '.js'],
			},
			module: {
				rules: [
					{
						test: /\.ts$/,
						exclude: [/node_modules/],
						use: [{
							loader: 'ts-loader',
						}],
					},
				],
			},
		},
	}))
};
```

(I found that `require('../../webpack.config')` is not the most intuitive design choice, because it would move the Cypress-only webpack configuration outside of the `cypress` dir, so for simplicity I just "inlined" the config, although one can argue that there is too much nesting now. Feel free to improve that.)

If I used the same layout as in the repo, then `cypress/plugins/cy-ts-preprocessor.js` would not be read at all, as nothing imports it.

With the content above, my TS files were interpreted as TypeScript. I had to install `@cypress/webpack-preprocessor` as well as `ts-loader` as dev dependencies.

Now the TypeScript transpiler would start processing the `.ts` files, but it fails to do without a proper `cypress/tsconfig.json`, so I used the one in this repo. Then my tests were successful even with TS syntax in them.
